### ISSUE TITLE
SimpleJSON read out of bounds - information leak

### ISSUE THREAD

**Issue by dzonerzy** [nbsdx/SimpleJSON#8]
While i was testing SimpleJSON security i found a crash during string parsing inside **parse_string**  function , below a screenshot.

<img width="979" alt="schermata 2017-02-07 alle 01 41 24" src="https://cloud.githubusercontent.com/assets/16840842/22672818/90d09e2c-ecd6-11e6-8fb3-718ba76d1103.png">

This seems to be an information leak bug since the parser will try to parse a string until it found a matching **"** character in order to close the string inside the object, so providing something similar will result in a read out of bounds!

Step to reproduce:

<img width="1007" alt="schermata 2017-02-07 alle 01 45 28" src="https://cloud.githubusercontent.com/assets/16840842/22672910/24489970-ecd7-11e6-8c44-632cfce7fae8.png">

Let me know if you need more information!

Regards,
Daniele Linguaglossa

**Comment by nbsdx** [comment]
Ohhhh cool :D I saw you had a JSON fuzzer on your github; is that how you found this?

Thanks for the heads up, I'll look into this, but I'm not sure how long it'll be. I'm in the middle of changing jobs and moving so I'm kinda swamped.

**Comment by nbsdx** [comment]
Yeah, taking a quick look at this, I'm going to probably need to rewrite the parser :/

**Comment by dzonerzy** [comment]
@nbsdx sure i found it using PyJFuzz :) so feel free to use it as a test suite during your development!
